db/rdbms/migration/0002_migrate_descriptor_to_extended_descriptor.go

```diff
@@ -0,0 +1,289 @@
+				return fmt.Errorf("could not retrieve test description from fetcher for jobID %d: %w", request.JobID, err)
+			}
+
+			// Check that the serialization of the steps retrieved by the test fetcher matches the steps
+			// stored in the DB. If that's not the case, then, just print a warning: the underlying test
+			/// might have changed.We go ahead anyway assuming assume the test name is still relevant.
+			stepDescFetcherJSON, err := json.Marshal(stepDescFetcher)
+			if err != nil {
+				log.Warningf("steps description (`%v`) fetched by test fetcher for job %d cannot be serialized: %v", stepDescFetcher, request.JobID, err)
+			}
+
+			stepDescDBJSON, err := json.Marshal(stepDesc)
+			if err != nil {
+				log.Warningf("steps description (`%v`) fetched from db for job %d cannot be serialized: %v", stepDesc, request.JobID, err)
+			}
+
+			if string(stepDescDBJSON) != string(stepDescFetcherJSON) {
+				log.Warningf("steps retrieved by test fetcher and from database do not match (`%v` != `%v`), test description might have changed", string(stepDescDBJSON), string(stepDescFetcherJSON))
+			}
+
+			newStepsDesc.TestName = name
+			extendedDescriptor.StepsDescriptors = append(extendedDescriptor.StepsDescriptors, newStepsDesc)
+		}
+
+		// Serialize job.ExtendedDescriptor
+		extendedDescriptorJSON, err := json.Marshal(extendedDescriptor)
+		if err != nil {
+			return fmt.Errorf("could not serialize extended descriptor for jobID %d (%+v): %w", request.JobID, extendedDescriptor, err)
+		}
+
+		insertStatement := "update jobs set extended_descriptor = ?  where job_id = ?"
+		log.Debugf("running insert statement: %s, descriptor: %s, jobID: %d", insertStatement, extendedDescriptorJSON, request.JobID)
+
+		insertStart := time.Now()
+		_, err = tx.Exec(insertStatement, extendedDescriptorJSON, request.JobID)
+		elapsed := time.Since(insertStart)
+		log.Debugf("insert statement executed in %.3f ms", ms(elapsed))
+
+		if err != nil {
+			return fmt.Errorf("could not store extended descriptor for job_id %d: %w", request.JobID, err)
+		}
+	}
+
+	elapsed := time.Since(start)
+	log.Debugf("completed migrating %d jobs in %.3f ms", len(requests), ms(elapsed))
+
+	return nil
+}
+
+// Up implements the forward migration
+func (m *DescriptorMigration) Up(tx *sql.Tx) error {
+
+	// Count how many entries we have in jobs table that we need to migrate. Split them into
+	// shards of size shardSize for migration. Can't be done online within a single transaction,
+	// as there cannot be two active queries on the same connection at the same time
+	// (see https://github.com/lib/pq/issues/81)
+
+	count := uint64(0)
+	m.log.Debugf("counting the number of jobs to migrate")
+	start := time.Now()
+	rows, err := tx.Query("select count(*) from jobs")
+	if err != nil {
+		return fmt.Errorf("could not fetch number of records to migrate: %v", err)
+	}
+	if !rows.Next() {
+		return fmt.Errorf("could not fetch number of records to migrate, at least one result from count(*) expected")
+	}
```

Comment:
**nitpicking:**

May be it makes sense to add `rows.Err()` to the error message.